src/common/components/layouts/pages/upload/Upload.js

```diff
@@ -1,39 +1,89 @@
-import React, { useMemo } from 'react';
+import React, { useEffect, useMemo } from 'react';
 import PropTypes from 'prop-types';
+import { ToggleButton, ToggleButtonGroup } from '@material-ui/lab';
+import Box from '@material-ui/core/Box';
 import { hasFilesAttached } from '../../../../store/pdfFiles/selectors';
 import { connect } from 'react-redux';
 import AppPages from '../../../AppPages';
 import Dropzone from './dropzone/Dropzone';
+import Inputzone from './inputzone/Inputzone';
 import WizardStep from '../../wizardStep/WizardStep';
 import PageNavigation from '../../../shared/pageNavigation/PageNavigation';
+import { setLink } from '../../../../store/pdfLink/actions';
+import { getFileLink, getFileError } from '../../../../store/pdfLink/selectors';
+import { isFileUploadMode } from '../../../../store/application/selectors';
+import { setFileUploadMode } from '../../../../store/application/actions';
 
-function Upload(props) {
-    const { filesAttached } = props;
-    const forwardButton = useMemo(
+import './Upload.scss';
+
+const ZONES = ['upload from computer', 'upload from web'];
+
+function Upload({ filesAttached, link, error, isFileUploadMode, setFileUploadMode }) {
+    const selectedZone = useMemo(() => ZONES[isFileUploadMode ? 0 : 1], [isFileUploadMode]);
+    const forwardButtonDrop = useMemo(
         () => ({
             label: 'Configure job',
             to: AppPages.SETTINGS,
             disabled: !filesAttached,
         }),
         [filesAttached]
     );
-
+    const forwardButtonInput = useMemo(
+        () => ({
+            label: 'Configure job',
+            to: AppPages.SETTINGS,
+            disabled: error,
+        }),
+        [error]
+    );
+    const buttons = useMemo(() => {
+        return ZONES.map(zone => (
+            <ToggleButton key={zone} value={zone}>
+                {zone}
+            </ToggleButton>
+        ));
+    }, []);
+    useEffect(() => {
```

Comment:
You don't need this `useEffect`, as you set link to the store each time the input changes